CONTRIBUTING.md

```diff
@@ -97,137 +76,310 @@ script before each commit and abort the commit on error. To
 disable the pre-commit hook run `pre-commit uninstall` from the
 root folder.
 
-### Testing Component Rendering and Functionality
+## Development Environments
 
-Components functionality is tested using the [Jest Testing Framework](https://jestjs.io/), and the
-[Enzyme Testing utility](https://enzymejs.github.io/enzyme/) and
-[Enzyme to JSON](https://github.com/adriantoine/enzyme-to-json), all of which are installed via package.json.
+The Retrospectives Extension can be built, developed and tested in several
+development environments. This section highlights three of the primary
+environments in order of relevance.
 
-#### Running Frontend Tests
+All of the development prerequisites, such as [Webpack](https://webpack.js.org/)
+and [NodeJS](https://nodejs.org/en/download/) are listed in the
+[Dockerfile](.devcontainer/Dockerfile). This file can be opened in a text
+editor and the install commands can be used to configure the prerequisites
+outside of a [dev container](#dev-containers).
 
-To execute tests, run the command the test command, which is `npm run test`, after installing the package.json file.
-This should run all of the tests that exist in files suffixed with `.test.tsx` inside of the [tests folder](RetrospectiveExtension.Frontend/components/__tests__). These
-tests currently ensure proper rendering (via snapshot tests) and expected output for basic use cases, and are grouped
-by component, inside of files that match the name of the component being tested. Please update these tests whenever
-necessary (including bug fixes) and include these tests in your pull requests.
+### Dev Containers
```

Comment:
Should we include the Docker Desktop install step? https://www.docker.com/products/docker-desktop

Reply:
great catch! I forgot about that as I installed Docker before we even started working on this project as I believe it's recommended in our onboarding docs.